### PR TITLE
Matched 2.0I and 2.0J, fixed matched object count calculation

### PR DESCRIPTION
**Pull request by Mr-Wiseguy**

### CODE REVIEW

File: src/os/getmemsize.c [hunk 1]
```diff
@@ -33,3 +35,19 @@ u32 osGetMemSize(void) {
...
+    do {
+        memsize += STEP;
+        memory = (u32*)(memsize + K1BASE);
+        memory[0] = 0x12345678;
+        memory[STEP / 4 - 1]  = 0x87654321;
+    } while (memory[0] == 0x12345678 && memory[STEP / 4 - 1] == 0x87654321);
+    return memsize;
+}
+
+#endif
```
Comment (hensldm) [inline src/os/getmemsize.c:53]: New line please :)

File: src/io/pfssetlabel.c [hunk 2]
```diff
@@ -21,11 +25,17 @@ s32 osPfsSetLabel(OSPfs* pfs, u8* label) {
         }
     }
 
+#if BUILD_VERSION >= VERSION_J
     if (pfs->activebank != 0) {
-        ret = (__osPfsSelectBank(pfs, 0));
+        ret = (SELECT_BANK(pfs, 0));
```
Comment (hensldm) [inline src/io/pfssetlabel.c:30]:
```suggestion
        ret = SELECT_BANK(pfs, 0);
```

File: src/io/pfsrepairid.c [hunk 3]
```diff
@@ -16,3 +18,56 @@ s32 osPfsRepairId(OSPfs* pfs) {
...
+    pfs->inode_start_page = pfs->banks * 2 + 3;
+    pfs->dir_size = 16;
+    pfs->inode_table = 8;
+    pfs->minode_table = (pfs->banks + 1) * PFS_ONE_PAGE;
+    pfs->dir_table = pfs->minode_table + pfs->banks * PFS_ONE_PAGE;
+    ERRCK(__osContRamRead(pfs->queue, pfs->channel, 7, pfs->label));
+    return 0;
+}
+
+#endif
```
Comment (hensldm) [inline src/io/pfsrepairid.c:73]: New line please

File: src/io/leodiskinit.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+#include "PR/os_internal.h"
+#include "PR/os_libc.h"
+#include "PR/rcp.h"
+#include "macros.h"
+
+OSPiHandle LeoDiskHandle ALIGNED(8);
+OSPiHandle *__osDiskHandle;
+
+OSPiHandle *osLeoDiskInit() {
+	u32 saveMask;
```
Comment (hensldm) [inline src/io/leodiskinit.c:11]: Hmm I think some of the code files you added used tabs intsead of 4 spaces. Not sure where they all are.

File: include/PR/os_motor.h [hunk 5]
```diff
@@ -65,9 +66,16 @@ extern s32 osMotorInit(OSMesgQueue *, OSPfs *, int);
...
 #define MOTOR_START		1
 #define MOTOR_STOP		0
+
+#if BUILD_VERSION >= VERSION_J
 #define	osMotorStart(x)		__osMotorAccess((x), MOTOR_START)
 #define	osMotorStop(x)		__osMotorAccess((x), MOTOR_STOP)
 extern s32 __osMotorAccess(OSPfs *, s32);
+#else
+extern s32 osMotorStart(OSPfs *);
+extern s32 osMotorStart(OSPfs *);
```
Comment (hensldm) [inline include/PR/os_motor.h:76]: Are these meant to be the same?

File: src/audio/synstartvoiceparam.c [hunk 6]
```diff
@@ -37,8 +41,15 @@ void alSynStartVoiceParams(ALSynth *s, ALVoice *v, ALWaveTable *w,
         update = (ALStartParamAlt *)__allocParam();
         ALFailIf(update == 0, ERR_ALSYN_NO_UPDATE);
 
-	if (fxmix > 127)
+#if BUILD_VERSION >= VERSION_J
+	    if (fxmix > 127) {
```
Comment (hensldm) [inline src/audio/synstartvoiceparam.c:45]: More tabs

File: src/io/cartrominit.c [hunk 7]
```diff
@@ -61,3 +63,33 @@ OSPiHandle* osCartRomInit(void) {
...
     return &__CartRomHandle;
 }
+#else
+
+OSPiHandle CartRomHandle ALIGNED(8);
+OSPiHandle* osCartRomInit(void) {
+    u32 domain = 0;
+    u32 saveMask;
+
+	if (CartRomHandle.baseAddress == PHYS_TO_K1(PI_DOM1_ADDR2))
```
Comment (hensldm) [inline src/io/cartrominit.c:73]: tabs

File: src/io/conteepread.c [hunk 8]
```diff
@@ -49,6 +50,32 @@ s32 osEepromRead(OSMesgQueue* mq, u8 address, u8* buffer) {
         __osSiRelAccess();
         return ret;
     }
+#else
+	if (ret != 0)
```
Comment (hensldm) [inline src/io/conteepread.c:54]: tab

**Review by hensldm — COMMENTED** [review]
Still seeing tabs in places. We may want to run a formatter.

**Comment by Mr-Wiseguy** [comment]
I do think a formatter would be a good idea, but there are a lot of files in this codebase that already have tab indentation and changing those would be outside the scope of this PR.

File: src/audio/reverb.c [hunk 9]
```diff
@@ -71,10 +70,10 @@ Acmd *alFxPull(void *filter, s16 *outp, s32 outCount, s32 sampleOffset,
 #ifdef AUD_PROFILE
     lastCnt[++cnt_index] = osGetCount();
 #endif
-    
-#ifdef _DEBUG
-    assert(source);
+#if BUILD_VERSION < VERSION_J
+#line 74
```
Comment (AngheloAlf) [inline src/audio/reverb.c:74]: Isn't this redundant? This `#line 74` is placed exactly in line 74, or does it apply to the very next statement?

Comment (Mr-Wiseguy) [reply src/audio/reverb.c:74]: Yup `#line` sets the line number of the next line, not the current one.

File: src/audio/synsetfxmix.c [hunk 10]
```diff
@@ -39,9 +39,18 @@ void alSynSetFXMix(ALSynth *synth, ALVoice *v, u8 fxmix)
...
             fxmix = 127;
-	update->data.i = fxmix;
+        }
+	    update->data.i = fxmix;
+#else
+        if (fxmix < 0) { // Not possible
+            update->data.i = -fxmix;
+        } else {
+            update->data.i = fxmix;
+        }
```
[the remark this answers is not in the document]
Comment (Mr-Wiseguy) [reply src/audio/synsetfxmix.c:52]: This is how it was in the original source, changing it to a ternary for an ABS macro breaks matching on libgultra_d.

File: src/audio/synstartvoiceparam.c [hunk 11]
```diff
@@ -37,8 +41,15 @@ void alSynStartVoiceParams(ALSynth *s, ALVoice *v, ALWaveTable *w,
...
 
-	if (fxmix > 127)
+#if BUILD_VERSION >= VERSION_J
+        if (fxmix > 127) {
             fxmix = 127;
+        }
+#else
+        if (fxmix < 0) { // Not possible
+            fxmix = -fxmix;
+        }
```
[the remark this answers is not in the document]
Comment (Mr-Wiseguy) [reply src/audio/synstartvoiceparam.c:51]: Ditto on it breaking matching.

**Review by AngheloAlf — APPROVED** [review]
LGTM

**Review by hensldm — APPROVED** [review]

**Pull request merged**